app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx

```diff
@@ -0,0 +1,262 @@
+const tooltipContext = classNames.bind(tooltipStyles);
+
+@injectIntl
+export class LaunchExecutionChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool.isRequired,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    height: 0,
+  };
+
+  state = {
+    isConfigReady: false,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeStatusChart);
+    this.getConfig();
+  }
+  componentWillUnmount() {
+    this.statusNode.removeEventListener('mousemove', this.setCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeStatusChart);
+  }
+
+  onStatusChartCreated = (chart, element) => {
+    this.statusChart = chart;
+    this.statusNode = element;
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    d3
+      .select(chart.element)
+      .select('.c3-chart-arcs-title')
+      .attr('dy', -15)
+      .append('tspan')
+      .attr('dy', 30)
+      .attr('x', 0)
+      .attr('fill', '#666')
+      .text('SUM');
+
+    this.statusNode.addEventListener('mousemove', this.setCoords);
+  };
+
+  onStatusMouseOut = () => {
+    this.statusChart.revert();
+  };
+
+  onStatusMouseOver = (id) => {
+    this.statusChart.focus(id);
+  };
+
+  onStatusClick = (id) => {
+    this.statusChart.toggle(id);
+  };
+
+  getConfig = () => {
+    const { widget, container } = this.props;
+    const statusChartData = {};
+    const statusChartDataOrdered = [];
+    const statusChartColors = {};
+    const values = widget.content.result[0].values;
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+    this.noAvailableData = false;
+
+    Object.keys(values).forEach((key) => {
+      if (key.indexOf('$executions$') > -1) {
+        const testStatus = key.split('$')[2].toUpperCase();
+        statusChartData[key] = +values[key];
+        statusChartColors[key] = COLORS[`COLOR_${testStatus}`];
+      }
+    });
+
+    statusChartData.statistics$executions$passed &&
+      statusChartDataOrdered.push([
+        'statistics$executions$passed',
+        statusChartData.statistics$executions$passed,
+      ]);
+    statusChartData.statistics$executions$failed &&
+      statusChartDataOrdered.push([
+        'statistics$executions$failed',
+        statusChartData.statistics$executions$failed,
+      ]);
+    statusChartData.statistics$executions$skipped &&
+      statusChartDataOrdered.push([
+        'statistics$executions$skipped',
+        statusChartData.statistics$executions$skipped,
+      ]);
+
+    if (
+      +statusChartDataOrdered.statistics$executions$total === 0 ||
+      !statusChartDataOrdered.length
+    ) {
+      this.noAvailableData = true;
+      return;
+    }
+
+    this.leftChartHidden = !statusChartDataOrdered.length;
```

Comment:
Should this widget know something about its position? )

Reply:
removed